openedx/core/djangoapps/content_libraries/views.py

```diff
@@ -178,6 +179,16 @@ class LibraryApiPagination(PageNumberPagination):
         ),
     ]
 
+    def get_paginated_response(self, data):
+        """Return a paginated style `Response` object for the given output data."""
+        return Response(OrderedDict([
+            ('count', self.page.paginator.count),
+            ('num_pages', self.page.paginator.num_pages),
```

Comment:
We should generally just be using https://github.com/openedx/edx-platform/blob/38320f8433427111c6cfb2bfaaa70aec5b989053/lms/envs/common.py#L3422 for all APIs, which [already includes `num_pages`](https://github.com/openedx/edx-drf-extensions/blob/85880da4c50fcfd7d3d5190444b848ae9f174968/edx_rest_framework_extensions/paginators.py#L28).

Reply:
Oh great! I was basing it off of what was done in courses v2 api in `HomePageCoursesPaginator`  https://github.com/openedx/edx-platform/blob/38320f8433427111c6cfb2bfaaa70aec5b989053/cms/djangoapps/contentstore/rest_api/v2/views/home.py#L20-L28

I'll swap it over to the one you linked, thanks for pointer!

Reply:
It should be the default so you can just remove any setting of pagination, and it should use that.

I don't know why the other API was written that way; maybe the author didn't know about our `DefaultPagination` class.

Reply:
I needed to swap out the `APIView -> GenericAPIView` and `ViewSet -> GenericViewSet` in order for the default pagination functionality to apply.